src/main/java/frc/robot/subsystems/drivetrain/Drivetrain.java

```diff
@@ -0,0 +1,172 @@
+
+public class Drivetrain extends SubsystemBase implements MoveableSubsystem {
+
+  private DifferentialDriveKinematics kinematics;
+  private DifferentialDriveOdometry odometry;
+
+  private WPI_TalonFX leftRearTalon;
+  private WPI_TalonFX leftMiddleTalon;
+  private WPI_TalonFX leftFrontTalon;
+  private WPI_TalonFX rightRearTalon;
+  private WPI_TalonFX rightMiddleTalon;
+  private WPI_TalonFX rightFrontTalon;
+  private WPI_TalonSRX rightEncoder;
+  private WPI_TalonSRX leftEncoder;
+  private Pigeon gyro;
+
+  private DifferentialDrive drivetrain;
+/**
+ * This is the subsystem of the drivetrain
+ */
+  public Drivetrain() {
+    leftRearTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_LEFT_REAR_TALON);
+    leftMiddleTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_LEFT_MIDDLE_TALON);
+    leftFrontTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_LEFT_FRONT_TALON);
+    rightRearTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_RIGHT_REAR_TALON);
+    rightMiddleTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_RIGHT_MIDDLE_TALON);
+    rightFrontTalon = new WPI_TalonFX(robotConstants.can.DRIVETRAIN_RIGHT_FRONT_TALON);
+    leftMiddleTalon.follow(leftFrontTalon);
+    leftRearTalon.follow(leftFrontTalon);
+    rightMiddleTalon.follow(rightFrontTalon);
+    rightRearTalon.follow(rightFrontTalon);
+    drivetrain = new DifferentialDrive(leftFrontTalon, rightFrontTalon);
+    drivetrain.setDeadband(0);
+    // TODO: set correct port for pigeon gyro.
+    gyro = new Pigeon(robotConstants.can.DRIVETRAIN_LEFT_REAR_TALON);
+    kinematics = new DifferentialDriveKinematics(robotConstants.drivetrainConstants.WHEEL_BASE_WIDTH);
+    odometry = new DifferentialDriveOdometry(Rotation2d.fromDegrees(getAngle()));
+    // TODO: set correct talons for encoders.
+    leftEncoder = new WPI_TalonSRX(robotConstants.can.TEMPORARY_TALON_FOR_LEFT_DRIVETRAIN_ENCODER);
+    rightEncoder = new WPI_TalonSRX(robotConstants.can.TEMPORARY_TALON_FOR_RIGHT_DRIVETRAIN_ENCODER);
+  }
+
+  public void periodic() {
+    odometry.update(Rotation2d.fromDegrees(getAngle()), getLeftDistance(), getRightDistance());
+  }
+
+  // Drive functions
+  public void arcadeDrive(double x, double y) {
+    drivetrain.arcadeDrive(x, y, false);
+  }
+
+  public void curvatureDrive(double x, double y, boolean quickTurn) {
+    drivetrain.curvatureDrive(x, y, quickTurn);
+  }
+
+  public void tankDrive(double leftSpeed, double rightSpeed) {
+    drivetrain.tankDrive(leftSpeed, rightSpeed, false);
+  }
+
+  public void voltageTankDrive(double leftVoltage, double rightVoltage) {
+    tankDrive(leftVoltage / RobotController.getBatteryVoltage(), rightVoltage / RobotController.getBatteryVoltage());
+  }
+
+  public void move(double power) {
+    arcadeDrive(0, power);
+  }
+
+  // Gyro functions
+  double getAngle() {
+    return Math.IEEEremainder(gyro.getAngle(), 360);
+  }
+
+  public double getRadianAngle() {
+    return Math.toRadians(gyro.getAngle() );
+  }
```

Comment:
return Math.toRadians(gyro.getAngle() );
to
    return Math.toRadians(gyro.getAngle());